packages/Maleo.js/src/client/client.tsx

```diff
@@ -19,12 +20,25 @@ export const init = async () => {
     const { data } = await ensureReady(routes, location.pathname, {});
 
     const wrapProps = await loadComponentProps(Wrap);
-    const appProps = await loadComponentProps(App);
+    const appInitialProps = await loadComponentProps(App);
+
+    const appProps = {
+      data,
+      routes,
+      location,
+      ...appInitialProps,
+      ...wrapProps,
+    };
+    const containerProps = {};
 
     const RenderApp = () => (
-      <Wrap {...wrapProps}>
-        <App data={data} routes={routes} location={location} {...appProps} {...wrapProps} />
-      </Wrap>
+      <Wrap
+        App={App}
+        Container={ContainerComponent}
```

Comment:
How if we put `Container` before `Wrap` since `Container` isn't extensible to the user?

Reply:
I think it is more versatile to have Container exposed to use, the user can choose what to do with the `Container` itself. And they are capable to add some provider above `Container`

Reply:
Ok, I got your point